src/game.cpp

```diff
@@ -1172,8 +1182,29 @@ void test_game() //!OCLINT tests may be many
   }
   #endif // FIX_ISSUE_524
 
-  //#define FIX_ISSUE_682
+  #define FIX_ISSUE_682
   #ifdef FIX_ISSUE_682
+  // A game has a function to check whether the game is over (check end game conditions, currently it only checks max time)
+  {
+    const int time_limit = 10;
+    const game_options g_options{3,
+                          false,
+                          get_random_kam(),
+                          get_random_kam(),
+                          get_random_kam(),
+                          environment_type(),
+                          time_limit
+                         };
+    game g{g_options};
+    for (int i = 0; i < time_limit; i++)
+      {
+        assert(!g.is_over());
+        g.increment_n_ticks();
+        g.check_over();
+      }
+    assert(g.is_over());
+  }
+
```

Comment:
I feel this test block is redundant with the next one. We don't need to explicitly test that `check_over()` exists: its existence (or something similar to it) is anyways required to make the test block below pass.

```suggestion
```